lib/i18n/hygiene/checks/missing_interpolation_variable_check.rb

```diff
@@ -0,0 +1,33 @@
+require 'i18n/hygiene/wrapper'
+require 'i18n/hygiene/checks/base'
+require 'i18n/hygiene/variable_checker'
+require 'i18n/hygiene/result'
+
+module I18n
+  module Hygiene
+    module Checks
+      class MissingInterpolationVariableCheck < Base
+        def run
+          puts "Checking for mismatching interpolation variables..."
+
+          wrapper = I18n::Hygiene::Wrapper.new
+
+          mismatched_variables = wrapper.keys_to_check.select do |key|
+            checker = I18n::Hygiene::VariableChecker.new(key, wrapper, config.locales)
+            checker.mismatch_details if checker.mismatched_variables_found?
+          end
+
+          mismatched_variables.each { |details| puts details }
+
+          puts "Finished checking.\n\n"
+
+          if mismatched_variables.any?
+            yield Result.new(:failure)
+          else
+            yield Result.new(:pass)
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
I agree, but considering a lot of the code that's doing the checks has similar problems, I felt like it was going to be a lot of extra work (which doesn't mean we shouldn't do it!).

For now I chose to just wrap them in a simple interface so we can refactor underneath as much as we want.